src/command_helpers.rs

```diff
@@ -301,14 +248,41 @@ pub(crate) fn command_add_output_file(
     }
 }
 
+#[cfg(feature = "parallel")]
+fn forward_available_stderr_as_warnings(
+    child_stderr_reader: &mut Option<BufReader<std::process::ChildStderr>>,
+) {
+    if let Some(child_stderr_reader) = child_stderr_reader.as_mut() {
+        if let Ok(available) = child_stderr_reader.fill_buf() {
+            let stdout = io::stdout();
+            let mut consumed = 0;
+            for line in available.split_inclusive(|&b| b == b'\n') {
+                // Only forward complete lines, leave the rest in the buffer.
+                if let Some((b'\n', line)) = line.split_last() {
+                    consumed += line.len() + 1;
+                    write_warning(&stdout, line);
```

Comment:
I think you also need to handle incomplete line on eof.

Tip: You could convert this function to an async one to make it easier to write, compile_objects does have some async support if you follow the example

Reply:
Ugh, I think the only way that I can handle that is to use `forward_all_stderr_as_warnings` once the process has exited, otherwise there's no way with `BufReader` to know if the underlying data source still has data left or has hit EOF without consuming all the bytes currently in the buffer.

Reply:
I also don't know how async would help here...

Reply:
async would make handling internal state easier, so instead of using BufReader you can just set to childstderr to non-blocking, keep reading from the child into a Vec and find newline until it returns `EWOULDBLOCK`, then yield control until it resumes.

Reply:
I don't see how to do this with the API exposed by `ChildStdErr`, unless you mean to convert it into its underlying fd or handle, which doesn't seem portable.

Reply:
That's true...it's only possible if you create the pipe manually.

Perhaps just add a detection of eof and then flush all pending data in the buffer?